scripts/BBoxManage.sh

```diff
@@ -0,0 +1,474 @@
+    then
+      failure
+    fi
+
+    chmod 660 $PASSWORD_CONFIG.original 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+
+    chown biblebox:biblebox $PASSWORD_CONFIG.original 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    fi
+  fi
+}
+
+function restore_password_config () {
+  # Backup the original configuration file
+  if [ -e "$PASSWORD_CONFIG.original" ]; then
+    if [ $DEBUG == 1 ]; then
+      echo 'Restoring $PASSWORD_CONFIG from $PASSWORD_CONFIG.original'
+    fi
+
+    cp $PASSWORD_CONFIG.original $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    else
+      reload_nginx
+    fi
+  fi
+}
+
+function reset () {
+  restore_password_config
+  restore_hostapd_config
+  restore_nginx_config
+}
+
+function success () {
+  echo $SUCCESS
+  exit 0;
+}
+
+function failure () {
+  echo $FAILURE
+  exit 1;
+}
+
+function reload_nginx () {
+  # Restart the nginx service
+  if [ $DEBUG == 1 ]; then
+    echo "Reloading nginx configuration"
+  fi
+
+  /etc/init.d/nginx reload
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function unmountusb () {
+  # Unmount usb
+  if [ $DEBUG == 1 ]; then
+    echo "Unmounting USB media"
+  fi
+
+  pumount /media/usb0 2>&1 | logger -t $(basename $0)
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doshutdown () {
+  # Shutdown
+  if [ $DEBUG == 1 ]; then
+    echo "Shutting down"
+  fi
+
+  /sbin/shutdown now 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doreboot () {
+  # Reboot
+  if [ $DEBUG == 1 ]; then
+    echo "Rebooting"
+  fi
+
+  /sbin/reboot 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function restart_hostapd () {
+  # Restart the hostapd service
+  if [ $DEBUG == 1 ]; then
+    echo "Restarting hostapd"
+  fi
+
+  /etc/init.d/hostapd restart
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function set_password () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing password value'
+    exit 1;
+  fi
+
+  backup_password_config
+
+  # Update the password property in the basicauth configuration file
+  if [ $DEBUG == 1 ]; then
+    echo "Updating password to '$val'"
+  fi
+
+  local new_hash=`echo $val | openssl passwd -apr1 -salt $PASSWORD_SALT -stdin`
+  echo "biblebox:$new_hash" > $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    reload_nginx
+  else
+    failure
+  fi
+}
+
+function get_hostname () {
+  local hostname=`grep 'bbox_server_name=' $NGINX_CONFIG`
+  local idx=`expr index "$hostname" =`
+
+  web_hostname="${hostname:$idx:${#hostname}-$idx}"
+
+  echo $web_hostname;
+}
+
+function set_hostname () {
```

Comment:
We haven't talked about what it means to change the hostname - if we're actually wanting to set the real hostname on the machine, then we could do that with the `hostname` command and have `server_name $hostname` (nginx seems to support this syntax)